scripts/retrieve_databundle_light.py

```diff
@@ -354,6 +356,73 @@ def download_and_unzip_direct(config, rootpath, hot_run=True, disable_progress=F
     return True
 
 
+def download_and_unzip_hydrobasins(
+    config, rootpath, hot_run=True, disable_progress=False
+):
+    """
+    download_and_unzip_basins(config, rootpath, dest_path, hot_run=True,
+    disable_progress=False)
+
+    Function to download and unzip the data for hydrobasins.
+
+    Inputs
+    ------
+    config : Dict
+        Configuration data for the category to download
+    rootpath : str
+        Absolute path of the repository
+    hot_run : Bool (default True)
+        When true the data are downloaded
+        When false, the workflow is run without downloading and unzipping
+    disable_progress : Bool (default False)
+        When true the progress bar to download data is disabled
+
+    Outputs
+    -------
+    True when download is successful, False otherwise
+    """
+    resource = config["category"]
+    url_templ = config["urls"]["hydrobasins"]
+    suffix_list = config["urls"]["suffixes"]
+
+    basins_fl = snakemake.config["renewable"]["hydro"]["resource"]["hydrobasins"]
+    level_pattern = r".*?lev(.*)_.*"
+    level_code = re.findall(level_pattern, basins_fl)
```

Comment:
Cool that you got how to identify the level from the path!
The level to be used later in the level_code to download should however be loaded from the config.yaml file.

Reply:
Have been trying to avoid changing the config :D
But you are absolutely right from the long-term perspective. Adjusted.